src/protocols/WebSocket.ts

```diff
@@ -209,15 +211,61 @@ export default class WebSocketProtocol extends BaseProtocolRealtime {
     });
   }
 
+  enableCookieSupport () {
+    if (typeof XMLHttpRequest === 'undefined') {
+      throw new Error('Support for cookie cannot be enabled outside of a browser');
+    }
+
+    super.enableCookieSupport();
+    this._httpProtocol = new HttpProtocol(
+      this.host, 
+      {
+        port: this.port,
+        ssl: this.ssl,
+      }
+    );
+    this._httpProtocol.enableCookieSupport();
+  }
+
   /**
    * Sends a payload to the connected server
    *
    * @param {Object} payload
    */
-  send (request: RequestPayload) {
-    if (this.client && this.client.readyState === this.client.OPEN) {
-      this.client.send(JSON.stringify(request));
+  send (request: RequestPayload, options: JSONObject = {}) {
+    if (!this.client || this.client.readyState !== this.client.OPEN) {
+      return;
+    }
+
+    if ( this.cookieSupport
+      && request.controller === 'auth'
+      && ( request.action === 'login'
+        || request.action === 'logout'
+        || request.action === 'refreshToken')
+    ) {
+      const verifiedRequest = this._httpProtocol.formatRequest(request, options);
+
+      if (verifiedRequest) {
+        if (this.client) {
+          this.client.close();
+        }
+        this.client = null;
+        this.clientDisconnected(); // Simulate a disconnection, this will enable offline queue and trigger realtime subscriptions backup
+
+        this._httpProtocol._sendHttpRequest(verifiedRequest.method, verifiedRequest.url, verifiedRequest.payload)
+          .then((response) => {
```

Comment:
Nit: coding standards
```suggestion
          .then(response => {
```